### PR TITLE
extract PDF BaseUrl From CxOriginUrl

### PR DESCRIPTION
**Pull request by swatipersistent**
Testing done on :
1. DC instance with /bamboo product name in URL
2. DC instance without /bamboo product name in URL
3. Bamboo server + remote agent + /bamboo product name in url
4. Bamboo server + remote agent + without /bamboo product name in url

### CODE REVIEW

File: src/main/java/com/cx/plugin/task/CheckmarxTask.java [hunk 1]
```diff
@@ -193,18 +193,18 @@ else if (scanSummary.hasErrors()) {
                     int buildNumber = buildContext.getParentBuildContext().getResultKey().getResultNumber();
                     String buildPath = buildContext.getPlanResultKey().getPlanKey().getKey()
                             .substring(buildContext.getPlanResultKey().getPlanKey().getKey().lastIndexOf("-") + 1);
-
+                    String pdfBaseUrl = extractPDFBaseUrlFromCxOriginUrl(config.getCxOriginUrl());
```
Comment (bebni) [inline src/main/java/com/cx/plugin/task/CheckmarxTask.java:196]: check the null condition for config.getCxOriginUrl()

File: src/main/java/com/cx/plugin/task/CheckmarxTask.java [hunk 2]
```diff
@@ -193,18 +193,18 @@ else if (scanSummary.hasErrors()) {
...
+                            sastResults.setSastPDFLink(pdfBaseUrl +"/browse/" + buildKey + "-" + buildNumber + "/artifact/shared/" + pdfArt.getName()
                                     + "/" + pdfName);
                         } else {
-                            sastResults.setSastPDFLink("/browse/" + buildKey + "-" + buildNumber + "/artifact/" + buildPath
+                            sastResults.setSastPDFLink(pdfBaseUrl +"/browse/" + buildKey + "-" + buildNumber + "/artifact/" + buildPath
                                     + "/" + pdfArt.getName() + "/" + pdfName);
                         }
                     } else {
-                        sastResults.setSastPDFLink("/browse/" + buildKey + "-" + buildNumber + "/artifact/" + buildPath
+                        sastResults.setSastPDFLink(pdfBaseUrl +"/browse/" + buildKey + "-" + buildNumber + "/artifact/" + buildPath
```
Comment (bebni) [inline src/main/java/com/cx/plugin/task/CheckmarxTask.java:207]: perform a testing for each block of if else condition.

File: src/main/java/com/cx/plugin/task/CheckmarxTask.java [hunk 3]
```diff
@@ -243,7 +243,29 @@ else if (scanSummary.hasErrors()) {
         }
     }
 
-    private ArtifactDefinitionContext getPDFArt(TaskContext taskContext) {
+    private String extractPDFBaseUrlFromCxOriginUrl(String cxOriginUrl) {
+    	   try {
+    	       int browseIndex = cxOriginUrl.indexOf("/browse");
```
Comment (bebni) [inline src/main/java/com/cx/plugin/task/CheckmarxTask.java:248]: before indexOf call check if cxOriginUrl contains "/browse" keyword

File: src/main/java/com/cx/plugin/task/CheckmarxTask.java (hunk 1 above)
Comment (swatipersistent) [reply src/main/java/com/cx/plugin/task/CheckmarxTask.java]: done

File: src/main/java/com/cx/plugin/task/CheckmarxTask.java (hunk 2 above)
Comment (swatipersistent) [reply src/main/java/com/cx/plugin/task/CheckmarxTask.java:210]: done

File: src/main/java/com/cx/plugin/task/CheckmarxTask.java (hunk 3 above)
Comment (swatipersistent) [reply src/main/java/com/cx/plugin/task/CheckmarxTask.java]: done

**Review by bebni — APPROVED** [review]

**Pull request merged**